backend/microlab/core.py

```diff
@@ -1,6 +1,21 @@
 """
 Contains function for starting up the microlab process
 """
+import logging
+import sys
+import time
+import threading
+import signal
+
+import hardware.devicelist
+import recipes.core
+import recipes.state
+
+from config import microlabConfig as config
+from hardware.core import MicroLabHardware
+
+HALT = threading.Event()
+MUTEX = threading.Lock()
```

Comment:
What's the reasoning behind moving these outside of the startMicrolabProcess function call? I think they're shared between the threads in either case?

Reply:
So the problem is that when the locks are instantiated within a method (or class *unless* it is as a class variable) you do get thread safe locks however you get different thread safe locks within each method call, I ran into this before when debugging a multiprocessing Lock issue. Ex: https://imgur.com/screenshot-MTl4AvQ

Reply:
Additionally given the structure of the code in this module I'm planning on moving it to a class as it is effectively one anyways with nested methods, at that point I will more than likely move the Locks into the class as class variables